packages/ens-utils/src/ethregistrar.ts

```diff
@@ -155,3 +179,284 @@ export function buildNFTRefFromENSName(
 
   return buildNFTRef(registrar.contract, token);
 }
+
+export const GRACE_PERIOD: Readonly<Duration> = buildDuration(
+  90n * SECONDS_PER_DAY.seconds,
+);
+export const TEMPORARY_PREMIUM_DAYS = 21n;
+
+export const TEMPORARY_PREMIUM_PERIOD: Readonly<Duration> = buildDuration(
+  TEMPORARY_PREMIUM_DAYS * SECONDS_PER_DAY.seconds,
+);
+
+export const DOMAIN_HAS_SPECIAL_PRICE_IF_LENGTH_EQUAL_OR_LESS_THAN = 5;
+
+/*
+  This interface defines data that is used to display the price of a domain
+  in the Ui. The reason we are separating this text in different fields is because:
+
+  1. We want to be able to display different texts depending on wether the price of
+  the domain is a premium price or not. In each one of these cases, the text displayed
+  is different.
+  2. Since the design for this data displaying is differently defined for the price field
+  and the descriptive text, we separate it so we can render these two fields separately in the
+  HTML that will be created inside the component. e.g. the price field is bold and the descriptive
+  text is not. Please refer to this Figma artboard for more details: https:/*www.figma.com/file/lZ8HZaBcfx1xfrgx7WOsB0/Namehash?type=design&node-id=12959-119258&mode=design&t=laEDaXW0rg9nIVn7-0
+*/
+export interface PriceDescription {
+  /* descriptiveTextBeginning references the text that is displayed before the price */
+  descriptiveTextBeginning: string;
+  /* pricePerYear is a string that represents: Price + "/ year" (e.g. "$5.99 / year") */
+  pricePerYearDescription: string;
+  /* descriptiveTextBeginning references the text that is displayed after the price */
+  descriptiveTextEnd: string;
+}
+
+export const getPriceDescription = (
```

Comment:
Do we have unit tests for this function? If not, can you please create them for us? That would be awesome. Thanks.

Reply:
We don't have! Do you want this during or after this PR ?

Reply:
@FrancoAguzzi We should do them as part of this PR. Thank you :+1:

Reply:
@FrancoAguzzi Sure sounds good :+1: I've created this Shortcut story to track this goal: https://app.shortcut.com/ps-web3/story/25596/add-tests-for-getpricedescription